### PR TITLE
feat: separate Passages tables

### PR DESCRIPTION
**Pull request by mlong93**
* modified passage tests
* passage listing consists of union of agent and source passages

**Please describe the purpose of this pull request.**
Is it to add a new feature? Is it to fix a bug?

**How to test**
How can we test your PR during review? What commands should we run? What outcomes should we expect?

**Have you tested this PR?**
Have you tested the latest commit on the PR? If so please provide outputs from your tests.

**Related issues or PRs**
Please link any related GitHub [issues](https://github.com/cpacker/Letta/issues) or [PRs](https://github.com/cpacker/Letta/pulls).

**Is your PR over 500 lines of code?**
If so, please break up your PR into multiple smaller PRs so that we can review them quickly, or provide justification for its length.

**Additional context**
Add any other context or screenshots about the PR here.

### CODE REVIEW

File: letta/orm/passage.py [hunk 1]
```diff
@@ -44,32 +50,63 @@ def process_result_value(self, value, dialect):
...
+    """Base class for all passage types with common fields"""
+    __abstract__ = True
     __pydantic_model__ = PydanticPassage
 
     id: Mapped[str] = mapped_column(primary_key=True, doc="Unique passage identifier")
     text: Mapped[str] = mapped_column(doc="Passage text content")
-    source_id: Mapped[Optional[str]] = mapped_column(nullable=True, doc="Source identifier")
     embedding_config: Mapped[dict] = mapped_column(EmbeddingConfigColumn, doc="Embedding configuration")
     metadata_: Mapped[dict] = mapped_column(JSON, doc="Additional metadata")
     created_at: Mapped[datetime] = mapped_column(DateTime(timezone=True), default=datetime.utcnow)
```
Comment (mattzh72) [inline letta/orm/passage.py:62]: this can be deleted

File: letta/agent.py [hunk 2]
```diff
@@ -93,7 +92,7 @@ def compile_memory_metadata_block(
         [
             f"### Memory [last modified: {timestamp_str}]",
             f"{message_manager.size(actor=actor, agent_id=agent_id) if message_manager else 0} previous messages between you and the user are stored in recall memory (use functions to access them)",
-            f"{passage_manager.size(actor=actor, agent_id=agent_id) if passage_manager else 0} total memories you created are stored in archival memory (use functions to access them)",
+            f"{agent_manager.passage_size(actor=actor, agent_id=agent_id) if agent_manager else 0} total memories you created are stored in archival memory (use functions to access them)",
```
Comment (mattzh72) [inline letta/agent.py:95]: why is agent_manager optional here?

File: letta/orm/passage.py [hunk 3]
```diff
@@ -1,49 +1,112 @@
...
+from letta.constants import MAX_EMBEDDING_DIM
+
 config = LettaConfig()
 
 if TYPE_CHECKING:
     from letta.orm.organization import Organization
+    from letta.orm.agent import Agent
+
+
+class CommonVector(TypeDecorator):
```
Comment (mattzh72) [inline letta/orm/passage.py:26]: i think this gets defined in `custom_columns.py`, let's moved shared definitions there

File: letta/services/agent_manager.py [hunk 4]
```diff
@@ -407,6 +409,260 @@ def detach_block_with_label(
...
+        start_date      : Optional[datetime] = None,
+        end_date        : Optional[datetime] = None,
+        cursor          : Optional[str] = None,
+        source_id       : Optional[str] = None,
+        embed_query     : bool = False,
+        ascending       : bool = True,
+        embedding_config: Optional[EmbeddingConfig] = None,
+        agent_only      : bool = False
+    ) -> int:
+        return len(
```
Comment (mattzh72) [inline letta/services/agent_manager.py:648]: rework this, limit should not be included here (doesn't make sense to call a size with a limit)

Comment (mattzh72) [reply letta/services/agent_manager.py:648]: should refactor the underlying queries with this and list_passage to be shared up until the limit

File: letta/services/agent_manager.py [hunk 5]
```diff
@@ -407,6 +409,263 @@ def detach_block_with_label(
...
+            )
+
+            # Add limit
+            if limit:
+                main_query = main_query.limit(limit)
+
+            # Execute query
+            results = list(session.execute(main_query))
+
+            passages = []
```
Comment (mattzh72) [inline letta/services/agent_manager.py:614]: can we do this outside of the db session?

Comment (mattzh72) [reply letta/services/agent_manager.py:614]: this should probably go into the custom `to_pydantic()`

**Review by mattzh72 — APPROVED** [review]

**Comment by mlong93** [comment]
TODO: unify `list_passages` logic (https://linear.app/letta/issue/LET-549/merge-agentmanagerlist-passages-logic-into-sqlalchemybaselist)

**Pull request merged**